MM-33103 Generate Support Packet Self-Managed (#4364)
mattermost/docs: source/cloud/cloud-administration/common-support-packet.rst

> Use the System Console to generate a Mattermost Support Packet that includes configuration information, logs, and data on external dependencies. Confidential data, such as passwords, are automatically stripped. 

Reviewer: Not sure if important to mention but it includes plugins installed too. I'm not sure if it falls under `external dependencies`.

.. note:: 

>   System Admins should always remove sensitive information before sending the support packet to Mattermost by email or via a support ticket. 

Reviewer 2: ```suggestion
  System Admins should always sanitize the contents to remove sensitive information before sending the support packet to Mattermost by email or via a support ticket.
```

To generate a support packet:

1. Go to System Console, then select **Commercial Support** from the **Main Menu**.
2. Select **Download Support Packet**. A zip file is downloaded to the local machine.

A Mattermost Support Packet contains the following five files:

> - mattermost.log

Reviewer 2: ```suggestion
- ``mattermost.log``
```

> - plugins.json

Reviewer 2: ```suggestion
- ``plugins.json``
```

> - sanitized_config.json

Reviewer 2: ```suggestion
- ``sanitized_config.json``
```

> - support_packet.yaml

Reviewer 2: ```suggestion
- ``support_packet.yaml``
```

> - warning.txt

Reviewer 2: ```suggestion
- ``warning.txt``
```

> - ``warning.txt``

Reviewer: `warning.txt` only appears if one of the files failed to be retrieved for whatever reason. It won't always be there if the above 4 files are retrieved without problem. Again not sure if it's important to point it out.

> You are notified if any packet files are unavailable during packet generation. See the ``warning.txt`` file for details.

Reviewer 2: ```suggestion
You'll be notified if any packet files are unavailable during packet generation. See the ``warning.txt`` file for details.
```

> 3. Sanitize configuration and log files in the Support Packet files to remove usernames, passwords, and LDAP groups. Replace these details with example strings that contain the same special characters if possible, as special characters are common causes of configuration errors.

Reviewer 2: We mention that passwords are automatically stripped in the opening paragraph, but mention here that they should be removed. I'm also wondering whether there's a specific method we recommend in terms of removing this information, or if it's something a System Admin would be inherently familiar with?

Author: @justinegeffen This is a great set of questions worth exploring with R&D further. If we can provide thought leadership and guidance to System Admins, they can complete these tasks faster and with greater confidence. 

@hahmadia  - Can you clarify what specifically gets sanitized when the support packet is generated, and what we recommend System Admins should do to further sanitize manually prior to sending the support packet to Mattermost? Thanks!

Reviewer: Sure thing! 

Nothing in `mattermost.log` , `plugins.json`, `support_packet` or `warning.txt` get sanitized or removed.

The following values in the `config.json` get sanitized:

- LdapSettings.BindPassword
- FileSettings.PublicLinkSalt
- FileSettings.AmazonS3SecretAccessKey
- EmailSettings.SMTPPassword
- GitLabSettings.Secret
- GoogleSettings.Secret 
- Office365Settings.Secret
- OpenIdSettings.Secret 
- SqlSettings.DataSource
- SqlSettings.AtRestEncryptKey 
- ElasticsearchSettings.Password
- All SqlSettings.DataSourceReplicas
- All SqlSettings.DataSourceSearchReplicas
- MessageExportSettings.GlobalRelaySettings.SmtpPassword
- ServiceSettings.GfycatApiSecret 
- ServiceSettings.SplitKey

If they like and feel this isn't enough, they can sanitize / remove whatever other data they want.

The  LDAP groups are not generated for us to view. Only LDAP Version and LDAP Vendor are generated if they are available in the `support_packet.yaml`

As for admins doing anything further, I think that's not required. I'd just recommend them to skim through the `plugins.json` and ensure their plugins don't have any sort of confidential information that needs to be hidden (like secret keys) because plugins are not sanitized but other than that I think they should be okay.

Author: Thank you for the great details, @hahmadia! I've requested a second review from you to ensure that I've captured these details accurately in the product documentation.

Use the System Console to generate a Mattermost Support Packet that includes configuration information, logs, plugin details, and data on external dependencies. Confidential data, such as passwords, are automatically stripped. 

.. note:: 

  When present, the following information is santized during packet generation: ``LdapSettings.BindPassword``, ``FileSettings.PublicLinkSalt``, ``FileSettings.AmazonS3SecretAccessKey``, ``EmailSettings.SMTPPassword``, ``GitLabSettings.Secret``, ``GoogleSettings.Secret``, ``Office365Settings.Secret``, ``OpenIdSettings.Secret``, ``SqlSettings.DataSource``, ``SqlSettings.AtRestEncryptKey``, ``ElasticsearchSettings.Password``, ``All SqlSettings.DataSourceReplicas``, ``All SqlSettings.DataSourceSearchReplicas``, ``MessageExportSettings.GlobalRelaySettings.SmtpPassword``, ``ServiceSettings.GfycatApiSecret``, and ``ServiceSettings.SplitKey``.

> To generate a support packet:

Reviewer 2: ```suggestion
To generate a Support Packet:
```

1. Go to System Console, then select **Commercial Support** from the **Main Menu**.
2. Select **Download Support Packet**. A zip file is downloaded to the local machine.

A Mattermost Support Packet can contain up to five files:

- ``mattermost.log``
- ``plugins.json``
- ``sanitized_config.json``
- ``support_packet.yaml``

> - ``warning.txt`` - present when issues are encountered during packet generation

Reviewer 2: ```suggestion
- ``warning.txt`` (present when issues are encountered during packet generation)
```

You'll be notified if any packet files are unavailable during packet generation. See the ``warning.txt`` file for details.

3. Plugins are not sanitized during packet generation. Sanitize any confidential details in the ``plugin.json`` file before sending the support packet to Mattermost. 

.. note::

>   When sanitizing support packet data, replace details with example strings that contain the same special characters if possible, as special characters are common causes of configuration errors.

Reviewer 2: ```suggestion
  When sanitizing Support Packet data, replace details with example strings that contain the same special characters if possible, as special characters are common causes of configuration errors.
```
